### PR TITLE
Check for gps errors

### PR DESCRIPTION
**Pull request by angelk**
Sometimes gps report wrong elevation.
These checks should eliminate fake points from elevation calculations and plots

Fix for https://github.com/trackhub/website/issues/137

### CODE REVIEW

File: src/Track/TwoPointsChecker/ElevationChecker.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public function __construct(float $maxAngle)
+    {
+        $this->maxAngle = $maxAngle;
+    }
+
+    public function isReal(Point $pointA, Point $pointB): bool
+    {
+        $distance = $pointA->distance($pointB);
+        $elevation = max($pointA->getElevation(), $pointB->getElevation()) - min($pointA->getElevation(), $pointB->getElevation());
+        $elevation = abs($elevation);
```
Comment (StefanMavrodiev) [inline src/Track/TwoPointsChecker/ElevationChecker.php]: Why not like this:
`$elevation = abs($pointA->getElevation() - $pointB->getElevation());` ?

File: src/Track/TwoPointsChecker/ElevationChecker.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        $atan = atan2($elevation, $distance);
+        $angle = rad2deg($atan);
```
Comment (StefanMavrodiev) [inline src/Track/TwoPointsChecker/ElevationChecker.php]: Can we make this one-liner?

File: src/Track/TwoPointsChecker/ElevationChecker.php (hunk 1 above)
Comment (angelk) [reply src/Track/TwoPointsChecker/ElevationChecker.php]: fixed

File: src/Track/TwoPointsChecker/ElevationChecker.php (hunk 2 above)
Comment (angelk) [reply src/Track/TwoPointsChecker/ElevationChecker.php]: For me `rad2deg(atan2($elevation, $distance))` looks more unreadable

**Pull request merged**